package/gluon-core/check_site.lua

```diff
@@ -39,3 +39,8 @@ for _, config in ipairs({'wifi24', 'wifi5'}) do
 end
 
 need_boolean('poe_passthrough', false)
+
+need_number('dns.cacheentries', false)
+need_string_array('dns.servers', false)
```

Comment:
While we're not always using these functions consistently, it would be better to wrap the `dns` checks in a
```lua
if need_table('dns', nil, false) then
```
block (and the same with `next_node`) to ensure that `site.dns` is a table if it exists.